gtsfm/loader/astronet_loader.py

```diff
@@ -0,0 +1,235 @@
+
+        Raises:
+            FileNotFoundError if image path does not exist.
+            ValueError if ground truth camera calibrations not provided.
+        """
+        self._use_gt_intrinsics = use_gt_intrinsics
+        self._use_gt_extrinsics = use_gt_extrinsics
+        self._use_gt_sfmtracks = use_gt_sfmtracks
+        self._max_frame_lookahead = max_frame_lookahead
+
+        # Use COLMAP model reader to load data and convert to GTSfM format
+        if Path(data_dir).exists():
+            _cameras, _images, _points3d = read_model(path=data_dir, ext=".bin")
+            self._calibrations, self._wTi_list, img_fnames, self._sfmtracks = self.colmap2gtsfm(
+                _cameras, _images, _points3d, load_sfmtracks=use_gt_sfmtracks
+            )
+
+        # Camera intrinsics are currently required due to absense of EXIF data and diffculty in approximating focal
+        # length (usually 10000 to 100000 pixels).
+        if self._calibrations is None:
+            raise ValueError("Camera calibrations cannot be None.")
+
+        if self._wTi_list is None:
+            self._use_gt_extrinsics = False
+
+        if self._sfmtracks is None:
+            self._use_gt_sfmtracks = False
+        self.num_sfmtracks = len(self._sfmtracks) if self._sfmtracks is not None else 0
+
+        # Prepare image paths
+        self._image_paths = []
+        for img_fname in img_fnames:
+            img_fpath = os.path.join(data_dir, "images", img_fname)
+            if not Path(img_fpath).exists():
+                raise FileNotFoundError(f"Could not locate image at {img_fpath}.")
+            self._image_paths.append(img_fpath)
+
+        self._num_imgs = len(self._image_paths)
+        logger.info("AstroNet loader found and loaded %d images and %d tracks.", self._num_imgs, self.num_sfmtracks)
+
+    @staticmethod
+    def colmap2gtsfm(
+        cameras: Dict[int, ColmapCamera],
+        images: Dict[int, ColmapImage],
+        points3D: Dict[int, ColmapPoint3D],
+        load_sfmtracks: Optional[bool] = False,
+    ) -> Tuple[List[Cal3Bundler], List[Pose3], List[str], List[Point3]]:
+        """Converts COLMAP-formatted variables to GTSfM format
+
+        Args:
+            cameras: dictionary of COLMAP-formatted Cameras
+            images: dictionary of COLMAP-formatted Images
+            points3D: dictionary of COLMAP-formatted Point3Ds
+            return_tracks (optional): whether or not to return tracks
+
+        Returns:
+            cameras_gtsfm: list of N camera calibrations corresponding to the N images in images_gtsfm
+            images_gtsfm: list of N camera poses when each image was taken
+            img_fnames: file names of images in images_gtsfm
+            sfmtracks_gtsfm: tracks of points in points3D
+        """
+        cameras_gtsfm, images_gtsfm, img_fnames, sfmtracks_gtsfm = None, None, None, None
+
+        # Note: Assumes input cameras use `PINHOLE` model
+        if len(images) > 0 and len(cameras) > 0:
```

Comment:
if len(images) == 0 or len(cameras) == 0 what is the expected behavior? should we throw an error?

i think we could avoid the nested logic by checking for that and throwing an exception right away

Reply:
Yes, I think throwing a runtime error if any of the Dicts are empty is a good idea.

Reply:
Done